lib/src/local_working_copy.rs

```diff
@@ -1619,13 +1620,17 @@ impl FileSnapshotter<'_> {
                 message: format!("Failed to read symlink {}", disk_path.display()),
                 err: err.into(),
             })?;
-            let str_target =
-                target
-                    .to_str()
-                    .ok_or_else(|| SnapshotError::InvalidUtf8SymlinkTarget {
-                        path: disk_path.to_path_buf(),
-                    })?;
-            Ok(self.store().write_symlink(path, str_target).await?)
+
+            // Convert from platform-specific PathBuf to a string.
+            let str_target = target
+                .iter()
+                .map(OsStr::to_str)
+                .collect::<Option<Vec<_>>>()
+                .ok_or_else(|| SnapshotError::InvalidUtf8SymlinkTarget {
+                    path: disk_path.to_path_buf(),
+                })?
+                .join("/");
```

Comment:
nit: maybe we can use `file_util::slash_path()`?

The reverse function can also be added to `file_util`, which is noop on non-Windows.